packages/kolibri-tools/lib/i18n/vue-intl-locale-data.js

```diff
@@ -0,0 +1,37 @@
+/*
+ * This is an auto-generated file, any manual edits will be overridden.
+ *
+ * To regenerate, see instructions here:
+ *   https://kolibri-dev.readthedocs.io/en/develop/references/i18n.html
+ *
+ * This file was generated by frontend_build/src/intl_code_gen.js
```

Comment:
We should probably update this comment :)

Reply:
> I think it should generally simplify a lot of the code too, as it's just a matter of doing a glob search for files, and then passing them to be AST parsed.

Unfortunately, how the string extraction works doesn't use the same AST parser as I used in the i18nSyncTransfer work (where I already took the glob approach). I thought it was a great idea before I started working - but as I got into it I quickly realized that it would be biting off more than is necessary to chew.

The AST parsers have minute details about them that are different - they call a node differently or structure the node metadata differently or they require that you use whatever weird specific options you need to accommodate features you're using or not using - so it isn't as easy as just using the existing ExtractStrings.js code and replacing webpack's build system w/ a different one.

Which is why I figured that adding some kind of condition to when/how that plugin is actually run would basically have the same ultimate outcome: working code with a single command (package.json script anyway) that extracts/profiles the messages.

Reply:
One thing I should add is that I went this route so that I could get it up and running sooner than later so that Radina can test the languages.

I'll need to test https://github.com/learningequality/studio/pull/2520 to see if it runs/builds without needing to actually use the newest `kolibri-tools` that is in this repo. I believe that the everything i18n that isn't to do with extraction/profiling/context transfer/Crowdin (aka `kolibri-tools`) is entirely local to Studio so if the translation JSONs are committed then the server should run.

In that case - then I can feel less pressure and put a big pretty webpack-less and env var-less bow on our cross-product i18n sometime between now and Thanksgiving.

Reply:
>  how the string extraction works doesn't use the same AST parser as I used in the i18nSyncTransfer

Ooof... sorry I should have caught this when you first did this work. Are you using esprima rather than espree? I had very specific issues with esprima that led me to use espree for the string extraction previously, so I guess I assumed you would face the same issues with esprima too...